client/utils/rafThrottle.js

```diff
@@ -0,0 +1,15 @@
+const rafThrottle = callback => {
```

Comment:
requestAnimationFrame should create a smoother animation than throttle and be more performant.

There is also an npm package that is similar if that is preferable:

https://www.npmjs.com/package/raf-throttle